src/dataflow-types/src/client.rs

```diff
@@ -205,47 +213,59 @@ impl RustType<ProtoComputeCommand> for ComputeCommand<mz_repr::Timestamp> {
                     })
                 }
-                ComputeCommand::Peek(peek) => Peek(peek.into_proto()),
-                ComputeCommand::CancelPeeks { uuids } => CancelPeeks(ProtoCancelPeeks {
+                ComputeCommandType::Peek(peek) => Peek(peek.into_proto()),
+                ComputeCommandType::CancelPeeks { uuids } => CancelPeeks(ProtoCancelPeeks {
                     uuids: uuids.into_proto(),
                 }),
             }),
+            additional_context: Some(AdditionalContext {
+                otel_ctx: self.otel_ctx.clone().into(),
+            }),
         }
     }
 
     fn from_proto(proto: ProtoComputeCommand) -> Result<Self, TryFromProtoError> {
         use proto_compute_command::Kind::*;
         use proto_compute_command::*;
-        match proto.kind {
-            Some(CreateInstance(config)) => Ok(ComputeCommand::CreateInstance(config.into_rust()?)),
-            Some(DropInstance(())) => Ok(ComputeCommand::DropInstance),
-            Some(CreateDataflows(ProtoCreateDataflows { dataflows })) => {
-                Ok(ComputeCommand::CreateDataflows(dataflows.into_rust()?))
-            }
-            Some(AllowCompaction(ProtoAllowCompaction { collections })) => {
-                Ok(ComputeCommand::AllowCompaction(collections.into_rust()?))
-            }
-            Some(Peek(peek)) => Ok(ComputeCommand::Peek(peek.into_rust()?)),
-            Some(CancelPeeks(ProtoCancelPeeks { uuids })) => Ok(ComputeCommand::CancelPeeks {
-                uuids: uuids.into_rust()?,
-            }),
-            None => Err(TryFromProtoError::missing_field(
-                "ProtoComputeCommand::kind",
-            )),
-        }
+        Ok(ComputeCommand {
+            cmd: match proto.kind {
+                Some(CreateInstance(config)) => {
+                    ComputeCommandType::CreateInstance(config.into_rust()?)
+                }
+                Some(DropInstance(())) => ComputeCommandType::DropInstance,
+                Some(CreateDataflows(ProtoCreateDataflows { dataflows })) => {
+                    ComputeCommandType::CreateDataflows(dataflows.into_rust()?)
+                }
+                Some(AllowCompaction(ProtoAllowCompaction { collections })) => {
+                    ComputeCommandType::AllowCompaction(collections.into_rust()?)
+                }
+                Some(Peek(peek)) => ComputeCommandType::Peek(peek.into_rust()?),
+                Some(CancelPeeks(ProtoCancelPeeks { uuids })) => ComputeCommandType::CancelPeeks {
+                    uuids: uuids.into_rust()?,
+                },
+                None => {
+                    return Err(TryFromProtoError::missing_field(
+                        "ProtoComputeCommandType::kind",
+                    ))
+                }
+            },
+            otel_ctx: proto
+                .additional_context
+                .map_or(OpenTelemetryContext::empty(), |ac| ac.otel_ctx.into()),
```

Comment:
I think the issue here is that we the Proto representation for an `OpenTelemetryContext` is less structured (`AdditionalContext` just wraps a `string ⇒ string` map). Consequently, it will be hard to classify errors occurring during deserialization into those that can be handled by the conversion method and those that should be propagated to the client.

Reply:
Another thing that I want to validate here: for command reconciliation, it's probably still sufficient to keep a durable log of `ComputeCommandType` rather than the now wrapping `ComputeCommand`. The `OpenTelemetryContext` seems like something ephemeral.

Reply:
> which is likely not possible

Wait, why not?

Reply:
Thanks, looks good to me.